src/Symfony/Component/Config/Builder/ArrayShapeGenerator.php

```diff
@@ -0,0 +1,158 @@
+{
+    public static function generate(ArrayNode $node): string
+    {
+        return self::prependPhpDocWithStar(self::doGeneratePhpDoc($node));
+    }
+
+    private static function doGeneratePhpDoc(NodeInterface $node, int $nestingLevel = 1): string
+    {
+        if (!$node instanceof ArrayNode) {
+            return $node->getName();
+        }
+
+        if ($node instanceof PrototypedArrayNode) {
+            $isHashmap = (bool) $node->getKeyAttribute();
+
+            $prototype = $node->getPrototype();
+            if ($prototype instanceof ArrayNode) {
+                return 'array<'.($isHashmap ? 'string, ' : '').self::doGeneratePhpDoc($prototype, $nestingLevel).'>';
+            }
+
+            return 'array<'.($isHashmap ? 'string, ' : '').self::handleScalarNode($prototype).'>';
+        }
+
+        if (!($children = $node->getChildren()) && !$node->getParent() instanceof PrototypedArrayNode) {
+            return 'array<array-key, mixed>';
+        }
+
+        $arrayShape = \sprintf("array{%s\n", self::generateInlinePhpDocForNode($node));
+
+        /** @var NodeInterface $child */
+        foreach ($children as $child) {
+            $arrayShape .= str_repeat(' ', $nestingLevel * 4).self::dumpNodeKey($child).': ';
+
+            if ($child instanceof PrototypedArrayNode) {
+                $isHashmap = (bool) $child->getKeyAttribute();
+
+                $arrayShape .= 'array<'.($isHashmap ? 'string, ' : '').self::handleNode($child->getPrototype(), $nestingLevel).'>';
+            } else {
+                $arrayShape .= self::handleNode($child, $nestingLevel);
+            }
+
+            $arrayShape .= \sprintf(",%s\n", !$child instanceof ArrayNode ? self::generateInlinePhpDocForNode($child) : '');
+        }
+
+        return $arrayShape.str_repeat(' ', 4 * ($nestingLevel - 1)).'}';
+    }
+
+    private static function dumpNodeKey(NodeInterface $node): string
+    {
+        $name = $node->getName();
+        $quoted = str_starts_with($name, '@')
+            || \in_array(strtolower($name), ['int', 'float', 'bool', 'null', 'scalar'], true)
+            || strpbrk($name, '\'"');
+
+        if ($quoted) {
+            $name = "'".addslashes($name)."'";
+        }
+
+        return $name.($node->isRequired() ? '' : '?');
+    }
+
+    private static function handleNumericNode(NumericNode $node): string
+    {
+        $min = $node->getMin() ?? 'min';
+        $max = $node->getMax() ?? 'max';
+
+        if ($node instanceof IntegerNode) {
+            return \sprintf('int<%s, %s>', $min, $max);
+        } elseif ($node instanceof FloatNode) {
+            return 'float';
+        }
+
+        return \sprintf('int<%s, %s>|float', $min, $max);
+    }
+
+    private static function prependPhpDocWithStar(string $shape): string
+    {
+        return str_replace("\n", "\n * ", $shape);
+    }
+
+    private static function generateInlinePhpDocForNode(BaseNode $node): string
+    {
+        $hasContent = false;
```

Comment:
needless var: $comment should start empty and be prefix only at the end, if not empty anymore
also, CR/LF in $comment should be replaced by a space so that we're sure it remains a oneliner

Reply:
It outputs something like this:

```
     *         strict_requirements?: string|int|float|bool, // set to true to throw an exception when a parameter does not match the requirements set to false to disable exceptions when a parameter does not match the requirements (and return null instead) set to null to disable parameter checks against requirements 'true' is the preferred configuration in development mode, while 'false' or 'null' might be preferred in production Default: true
```

Maybe we should replace `/\s/` with `; ` instead?

Reply:
It currently outputs something like this for `framework`:

```
     *         strict_requirements?: string|int|float|bool, // set to true to throw an exception when a parameter does not match the requirements set to false to disable exceptions when a parameter does not match the requirements (and return null instead) set to null to disable parameter checks against requirements 'true' is the preferred configuration in development mode, while 'false' or 'null' might be preferred in production Default: true
```

Maybe we could instead replace `\r` and `\n` with `; `?

Reply:
Hmm no, it seems it a particular case. Other places finish the info with a `.`, so maybe we should adjust the info of this particular node.